packages/aws-cdk-lib/core/lib/helpers-internal/cfn-parse.ts

```diff
@@ -853,3 +857,55 @@ export class CfnParser {
     return this.options.parameters || {};
   }
 }
+
+class ObjectParser<T extends object> {
+  private readonly parsed: Record<string, unknown> = {};
+  private readonly unparsed: Record<string, unknown> = {};
+
+  constructor(input: Record<string, unknown>) {
+    this.unparsed = { ...input };
+  }
+
+  // eslint-disable-next-line max-len
+  public parse<K extends keyof T>(targetKey: K, key: string, parser: (x: any) => T[K] | FromCloudFormationResult<T[K] | IResolvable>) {
```

Comment:
```suggestion
  public parse<K extends keyof T>(targetKey: K, key: Capitalize<K>, parser: (x: any) => T[K] | FromCloudFormationResult<T[K] | IResolvable>) {
```

Reply:
But I agree with @otaviomacedo A simple `firstCharUpper` would be even better.

Reply:
I didn't because we can't assume this is true in general. But I guess this is on a small enough domain that it's always true here.

Reply:
Didn't know about `Capitalize` btw. That's neat.